chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,317 @@
+        if getattr(self, "vault_info", None):
+            return True
+        return False
+
+    async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
+        raise NotImplementedError("vault wallet")
+
+    def handle_own_derivation(self) -> bool:
+        return True
+
+    def get_recovery_info(self) -> Tuple[Optional[G1Element], Optional[uint64]]:
+        if self.vault_info.is_recoverable:
+            return self.recovery_info.bls_pk, self.recovery_info.timelock
+        return None, None
+
+    def derivation_for_index(self, index: int) -> List[DerivationRecord]:
+        hidden_puzzle = get_vault_hidden_puzzle_with_index(uint32(index))
+        hidden_puzzle_hash = hidden_puzzle.get_tree_hash()
+        bls_pk, timelock = self.get_recovery_info()
+        inner_puzzle_hash = get_vault_inner_puzzle_hash(
+            self.vault_info.pubkey,
+            DEFAULT_CONSTANTS.GENESIS_CHALLENGE,
+            hidden_puzzle_hash,
+            bls_pk,
+            timelock,
+        )
+        record = DerivationRecord(
+            uint32(index), inner_puzzle_hash, self.vault_info.pubkey, self.type(), self.id(), False
+        )
+        return [record]
+
+    async def create_recovery_spends(self) -> Tuple[CoinSpend, CoinSpend]:
+        """
+        Returns two spendbundles
+        1. The spend recovering the vault which can be taken to the appropriate BLS wallet for signing
+        2. The spend that completes the recovery after the timelock has elapsed
+        """
+        assert self.vault_info.is_recoverable
+        wallet_node: Any = self.wallet_state_manager.wallet_node
+        peer = wallet_node.get_full_node_peer()
+        assert peer is not None
+        # 1. Generate the recovery spend
+        # Get the current vault coin, ensure it's unspent
+        vault_coin = self.vault_info.coin
+        amount = uint64(self.vault_info.coin.amount)
+        vault_coin_state = (await wallet_node.get_coin_state([vault_coin.name()], peer))[0]
+        assert vault_coin_state.spent_height is None
+        # Generate the current inner puzzle
+        inner_puzzle = get_vault_inner_puzzle(
+            self.vault_info.pubkey,
+            DEFAULT_CONSTANTS.GENESIS_CHALLENGE,
+            self.vault_info.hidden_puzzle_hash,
+            self.recovery_info.bls_pk,
+            self.recovery_info.timelock,
+        )
+        assert inner_puzzle.get_tree_hash() == self.vault_info.inner_puzzle_hash
+
+        secp_puzzle_hash = construct_p2_delegated_secp(
+            self.vault_info.pubkey,
+            DEFAULT_CONSTANTS.GENESIS_CHALLENGE,
+            self.vault_info.hidden_puzzle_hash,
+        ).get_tree_hash()
+
+        recovery_puzzle = get_recovery_puzzle(
+            secp_puzzle_hash, self.recovery_info.bls_pk, self.recovery_info.timelock, amount
+        )
+        recovery_puzzle_hash = recovery_puzzle.get_tree_hash()
+        # TODO: Once we have get_wallet_balance we can use it for the amount field in the solution
```

Comment:
I feel like the amount of the vault will always be one in standard operation.  I'm not saying hardcode 1 here, but use `vault_info.coin` maybe and I think `get_wallet_balance` is irrelevant.